src/realm/sync/subscriptions.hpp

```diff
@@ -349,6 +349,15 @@ class SubscriptionStore : public std::enable_shared_from_this<SubscriptionStore>
                                                                  DB::version_type after_client_version) const;
     std::vector<SubscriptionSet> get_pending_subscriptions() const;
 
+    // Notify all subscription state change notification handlers on this subscription store with the
+    // provided Status - this does not change the state of any pending subscriptions.
+    // Must be called from the event loop thread.
+    void notify_all_state_change_notifications(Status error);
+
+    // Reset SubscriptionStore and erase all current subscriptions and supersede any pending
+    // subscriptions. Must be called from the event loop thread.
+    void terminate();
```

Comment:
Since this is calling the pending subscription notifications, I was worried that there might be side-effects in some of the handlers that require it to be on the event loop thread. In addition, `terminate()` will only be called on the event loop thread anyways as the result of receiving the "RevertToPBS" client reset action.

Reply:
How do we ensure these are called from the event loop? I guess that's the case because they are called only by the sync client (so maybe this should be documented instead)